airflow/contrib/operators/gcp_compute_operator.py

```diff
@@ -181,3 +195,241 @@ def execute(self, context):
+
+    def execute(self, context):
+        self._validate_all_body_fields()
+        try:
+            # Idempotence check (sort of) - we want to check if the new template
+            # is already created and if is, then we assume it was created by previous run
+            # of CopyTemplate operator - we do not check if content of the template
+            # is as expected. Templates are immutable so we cannot update it anyway
+            # and deleting/recreating is not worth the hassle especially
+            # that we cannot delete template if it is already used in some Instance
+            # Group Manager. We assume success if the template is simply present
+            existing_template = self._hook.get_instance_template(
+                project_id=self.project_id, resource_id=self.body_patch['name'])
+            self.log.info("The {} template already existed. It was likely "
+                          "created by previous run of the operator. Assuming success.")
+            return existing_template
+        except HttpError as e:
+            # We actually expect to get 404 / Not Found here as the template should
+            # not yet exist
+            if not e.resp.status == 404:
+                raise e
+        old_body = self._hook.get_instance_template(project_id=self.project_id,
+                                                    resource_id=self.resource_id)
+        new_body = deepcopy(old_body)
+        self._field_sanitizer.sanitize(new_body)
+        new_body = merge(new_body, self.body_patch)
+        self.log.info("Calling insert instance template with updated body: {}".
+                      format(new_body))
+        self._hook.insert_instance_template(project_id=self.project_id,
+                                            body=new_body,
+                                            request_id=self.request_id)
+        return self._hook.get_instance_template(project_id=self.project_id,
+                                                resource_id=self.body_patch['name'])
+
+
+class GceInstanceGroupManagerUpdateTemplateOperator(GceBaseOperator):
+    """
+    Patches the Instance Group Manager, replacing source template URL with the
+    destination one. API V1 does not have update/patch operations for Instance
+    Group Manager, so you must use beta or newer API version. Beta is the default.
+
+    :param project_id: Google Cloud Platform Project ID where the Compute Engine
+        instance exists.
+    :type project_id: str
+    :param resource_id: Name of the Instance Group Manager
+    :type resource_id: str
+    :param zone: Google Cloud Platform zone where the Instance Group Manager exists.
+    :type zone: str
+    :param request_id: Optional, unique request_id that you might add to achieve
+           full idempotence (for example when client call times out repeating the request
+           with the same request id will not create a new instance template again).
```

Comment:
Can we please make the indentation for all the docstrings in this PR consistent? Only one indent or tab is required.

Reply:
Yep. Reviewed and fixed everywhere.